### DISCUSSION TITLE
Error upgrading from 4.0.11 to 4.1.0

### DISCUSSION THREAD

**Discussion by FrankPGH** [netbox-community/netbox#17388] [Help Wanted!]
Stock install and never had issues with previous upgrades. Upgrade to 4.1.0 fails at database migration. I don't *think* I have any plugins or extras, and no customizations, let alone ones that would have affected DB access privileges.

```
Running migrations:
  Applying core.0011_move_objectchange... OK
  Applying extras.0117_move_objectchange...Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 103, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.12/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
psycopg.errors.InsufficientPrivilege: permission denied for schema public

The above exception was the direct cause of the following exception:
<follow-on errors omitted>
```
Followed suggestion from [this thread from last year](https://github.com/netbox-community/netbox/discussions/11811)

```
postgres=# ALTER DATABASE netbox OWNER TO netbox;
ALTER DATABASE
postgres=# \connect netbox;
postgres=# GRANT ALL ON SCHEMA public TO netbox;
GRANT

```

Re-running upgrade still fails, but differently:
```
Running migrations:
  Applying extras.0117_move_objectchange...Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/netbox/venv/lib/python3.12/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
psycopg.errors.UndefinedTable: relation "extras_objectchange" does not exist
LINE 1: ...ct_type_id", "extras_objectchange"."user_id" FROM "extras_ob...
                                                             ^

The above exception was the direct cause of the following exception:
```

**Comment by quantum909** [comment]
Hello,

had they exactly same problem as you. Also the suggestion from https://github.com/netbox-community/netbox/discussions/11811 and https://github.com/netbox-community/netbox/discussions/17382 did not help.

Did a dump from PostgreSQL 16.3 and a fresh install on Ubuntu 24.04.1 LTS and PostgreSQL 16.4. The backup import into the new PostgreSQL netbox database was smooth.

**Reply by FrankPGH** [reply]
Already on 16.4:

> Ubuntu 24.04.1 LTS
> psql (PostgreSQL) 16.4 (Ubuntu 16.4-0ubuntu0.24.04.2)

**Reply by romanpilipovskiy** [reply]
@quantum909 Same problem when migrate from `v4.0-2.9.1` to `v4.1-3.0.1`:
```
django.db.utils.ProgrammingError: relation "extras_objectchange" does not exist
 LINE 1: ...ct_type_id", "extras_objectchange"."user_id" FROM "extras_ob...
```
Suggestion from https://github.com/netbox-community/netbox/discussions/11811 and https://github.com/netbox-community/netbox/discussions/17382 did not help.

**Reply by lopar** [reply]
I tried the same, made new instance, made fresh install, the dropped netbox DB, created new and wrote a dump from last worked version. It didn't help at all. Got this on new instance:

```
django.db.utils.ProgrammingError: relation "extras_objectchange" does not exist
LINE 1: ...ct_type_id", "extras_objectchange"."user_id" FROM "extras_ob...
                                                             ^
  Applying extras.0117_move_objectchange...
```

Exact the same as on old one.

**Comment by avbor** [comment]
Full log with this error from postgresql side - https://github.com/netbox-community/netbox/discussions/17382#discussioncomment-10570220

**Comment by avbor** [comment]
Rised an issue https://github.com/netbox-community/netbox/issues/17401, everyone please check in there.

**Comment by amhn** [comment]
It seems the failing migration extras.0117_move_objectchange fails, if core.0011_move_objectchange is not executed in the same run.

Try
```
delete from django_migrations WHERE name='0011_move_objectchange';
```

The 0011_move_objectchange is a state only migration that does not change anything in the database. On the first run there was a permission problem on the database, which caused extras.0117_move_objectchange to fail, but core.0011_move_objectchange was executed and recorded in the migrations table, because it does not change the database structure.

**Reply by fischbyte** [reply]
> delete from django_migrations WHERE name='0011_move_objectchange';

This fixed it. Thank you so much! :)

**Comment by avbor** [comment]
@amhn 
Great suggestion!
But in my case it was also necessary to delete the redis cache, for some reason I didn't see these errors before...

After:
```
root@srv02:/opt/docker-compose/netbox-docker# docker exec -it netbox-db psql --username=netbox
psql (16.4)
Type "help" for help.

netbox=# delete from django_migrations WHERE name='0011_move_objectchange';
DELETE 1
```
I saw in log another error about redis cache:

```
netbox               | ⚙️ Applying database migrations
netbox               | 🧬 loaded config '/etc/netbox/config/configuration.py'
netbox               | 🧬 loaded config '/etc/netbox/config/custom_config.py'
netbox               | 🧬 loaded config '/etc/netbox/config/extra.py'
netbox               | 🧬 loaded config '/etc/netbox/config/logging.py'
netbox               | 🧬 loaded config '/etc/netbox/config/plugins.py'
netbox               | Operations to perform:
netbox               |   Apply all migrations: account, auth, circuits, contenttypes, core, dcim, django_rq, extras, ipam, netbox_dns, sessions, social_django, taggit, tenancy, users, virtualization, vpn, wireless
netbox               | Running migrations:
netbox               |   Applying core.0011_move_objectchange... OK
netbox               | Traceback (most recent call last):
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 29, in _decorator
netbox               |     return method(self, *args, **kwargs)
netbox               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 99, in _get
netbox               |     return self.client.get(key, default=default, version=version, client=client)
netbox               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/client/default.py", line 260, in get
netbox               |     raise ConnectionInterrupted(connection=client) from e
netbox               | django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Error -2 connecting to redis-cache:6379. Name or service not known.
netbox               |
netbox               | During handling of the above exception, another exception occurred:
netbox               |
netbox               | Traceback (most recent call last):
netbox               |   File "/opt/netbox/netbox/./manage.py", line 10, in <module>
netbox               |     execute_from_command_line(sys.argv)
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
netbox               |     utility.execute()
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/__init__.py", line 436, in execute
netbox               |     self.fetch_command(subcommand).run_from_argv(self.argv)
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/base.py", line 413, in run_from_argv
netbox               |     self.execute(*args, **cmd_options)
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/base.py", line 459, in execute
netbox               |     output = self.handle(*args, **options)
netbox               |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/base.py", line 107, in wrapper
netbox               |     res = handle_func(*args, **kwargs)
netbox               |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/commands/migrate.py", line 356, in handle
netbox               |     post_migrate_state = executor.migrate(
netbox               |                          ^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 135, in migrate
netbox               |     state = self._migrate_all_forwards(
netbox               |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
netbox               |     state = self.apply_migration(
netbox               |             ^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
netbox               |     state = migration.apply(state, schema_editor)
netbox               |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/migration.py", line 132, in apply
netbox               |     operation.database_forwards(
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards
netbox               |     self.code(from_state.apps, schema_editor)
netbox               |   File "/opt/netbox/netbox/extras/migrations/0117_move_objectchange.py", line 8, in update_content_types
netbox               |     ContentType.objects.filter(app_label='core', model='objectchange').delete()
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/models/query.py", line 1188, in delete
netbox               |     deleted, _rows_count = collector.delete()
netbox               |                            ^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/models/deletion.py", line 463, in delete
netbox               |     signals.pre_delete.send(
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/dispatch/dispatcher.py", line 189, in send
netbox               |     response = receiver(signal=self, sender=sender, **named)
netbox               |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/netbox/core/signals.py", line 121, in handle_deleted_object
netbox               |     validators = get_config().PROTECTION_RULES.get(model_name, [])
netbox               |                  ^^^^^^^^^^^^
netbox               |   File "/opt/netbox/netbox/netbox/config/__init__.py", line 28, in get_config
netbox               |     _thread_locals.config = Config()
netbox               |                             ^^^^^^^^
netbox               |   File "/opt/netbox/netbox/netbox/config/__init__.py", line 48, in __init__
netbox               |     self._populate_from_cache()
netbox               |   File "/opt/netbox/netbox/netbox/config/__init__.py", line 71, in _populate_from_cache
netbox               |     self.config = cache.get('config') or {}
netbox               |                   ^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 92, in get
netbox               |     value = self._get(key, default, version, client)
netbox               |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 36, in _decorator
netbox               |     raise e.__cause__
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/client/default.py", line 258, in get
netbox               |     value = client.get(key)
netbox               |             ^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/commands/core.py", line 1821, in get
netbox               |     return self.execute_command("GET", name)
netbox               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/client.py", line 545, in execute_command
netbox               |     conn = self.connection or pool.get_connection(command_name, **options)
netbox               |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/connection.py", line 1074, in get_connection
netbox               |     connection.connect()
netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/connection.py", line 283, in connect
netbox               |     raise ConnectionError(self._error_message(e))
netbox               | redis.exceptions.ConnectionError: Error -2 connecting to redis-cache:6379. Name or service not known.
netbox               |   Applying extras.0117_move_objectchange...
```

and this errors from redis and redis-cache containers:
```
netbox-redis         | 1:M 08 Sep 2024 16:11:22.284 * Reading RDB base file on AOF loading...
netbox-redis         | 1:M 08 Sep 2024 16:11:22.284 # Can't handle RDB format version 12
netbox-redis         | 1:M 08 Sep 2024 16:11:22.284 # Error reading the RDB base file appendonly.aof.65.base.rdb, AOF loading aborted
```

After purge volumes from this containers and and re-removal 0011_move_objectchange from django_migrations netbox is up and running:

```
netbox               | Running migrations:
netbox               |   Applying core.0011_move_objectchange... OK
netbox               |   Applying extras.0117_move_objectchange... OK
netbox               |   Applying extras.0118_customfield_uniqueness... OK
netbox               |   Applying extras.0119_notifications... OK
netbox               |   Applying circuits.0044_circuit_groups... OK
netbox               |   Applying core.0012_job_object_type_optional... OK
netbox               |   Applying extras.0120_eventrule_event_types... OK
netbox               |   Applying extras.0121_customfield_related_object_filter... OK
netbox               |   Applying dcim.0188_racktype... OK
netbox               |   Applying dcim.0189_moduletype_rack_airflow... OK
netbox               |   Applying dcim.0190_nested_modules... OK
netbox               |   Applying ipam.0070_vlangroup_vlan_id_ranges... OK
netbox               |   Applying virtualization.0039_virtualmachine_serial_number... OK
netbox               |   Applying virtualization.0040_convert_disk_size... OK
netbox               |   Applying wireless.0009_wirelesslink_distance... OK
```

Thank you very much!

**Reply by amhn** [reply]
Thank you. I was still looking for the initial error on the database side. Removing the redis volumes before starting the new containers fixed the upgrade for me too.

**Reply by rock7632** [reply]
> After purge volumes from this containers and and re-removal 0011_move_objectchange from django_migrations netbox is up and running:
> ```
> 
> Thank you very much!

can you please tell me how you purged volumes from the redis container?

**Reply by avbor** [reply]
> can you please tell me how you purged volumes from the redis container?

Something like this...

find volume name:
`docker volume ls`

delete volume:
`docker volume rm <volume_name>`

**Reply by tyler-jurgens-s2d** [reply]
> @amhn Great suggestion! But in my case it was also necessary to delete the redis cache, for some reason I didn't see these errors before...
> 
> After:
> 
> ```
> root@srv02:/opt/docker-compose/netbox-docker# docker exec -it netbox-db psql --username=netbox
> psql (16.4)
> Type "help" for help.
> 
> netbox=# delete from django_migrations WHERE name='0011_move_objectchange';
> DELETE 1
> ```
> 
> I saw in log another error about redis cache:
> 
> ```
> netbox               | ⚙️ Applying database migrations
> netbox               | 🧬 loaded config '/etc/netbox/config/configuration.py'
> netbox               | 🧬 loaded config '/etc/netbox/config/custom_config.py'
> netbox               | 🧬 loaded config '/etc/netbox/config/extra.py'
> netbox               | 🧬 loaded config '/etc/netbox/config/logging.py'
> netbox               | 🧬 loaded config '/etc/netbox/config/plugins.py'
> netbox               | Operations to perform:
> netbox               |   Apply all migrations: account, auth, circuits, contenttypes, core, dcim, django_rq, extras, ipam, netbox_dns, sessions, social_django, taggit, tenancy, users, virtualization, vpn, wireless
> netbox               | Running migrations:
> netbox               |   Applying core.0011_move_objectchange... OK
> netbox               | Traceback (most recent call last):
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 29, in _decorator
> netbox               |     return method(self, *args, **kwargs)
> netbox               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 99, in _get
> netbox               |     return self.client.get(key, default=default, version=version, client=client)
> netbox               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/client/default.py", line 260, in get
> netbox               |     raise ConnectionInterrupted(connection=client) from e
> netbox               | django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Error -2 connecting to redis-cache:6379. Name or service not known.
> netbox               |
> netbox               | During handling of the above exception, another exception occurred:
> netbox               |
> netbox               | Traceback (most recent call last):
> netbox               |   File "/opt/netbox/netbox/./manage.py", line 10, in <module>
> netbox               |     execute_from_command_line(sys.argv)
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
> netbox               |     utility.execute()
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/__init__.py", line 436, in execute
> netbox               |     self.fetch_command(subcommand).run_from_argv(self.argv)
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/base.py", line 413, in run_from_argv
> netbox               |     self.execute(*args, **cmd_options)
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/base.py", line 459, in execute
> netbox               |     output = self.handle(*args, **options)
> netbox               |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/base.py", line 107, in wrapper
> netbox               |     res = handle_func(*args, **kwargs)
> netbox               |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/core/management/commands/migrate.py", line 356, in handle
> netbox               |     post_migrate_state = executor.migrate(
> netbox               |                          ^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 135, in migrate
> netbox               |     state = self._migrate_all_forwards(
> netbox               |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
> netbox               |     state = self.apply_migration(
> netbox               |             ^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
> netbox               |     state = migration.apply(state, schema_editor)
> netbox               |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/migration.py", line 132, in apply
> netbox               |     operation.database_forwards(
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards
> netbox               |     self.code(from_state.apps, schema_editor)
> netbox               |   File "/opt/netbox/netbox/extras/migrations/0117_move_objectchange.py", line 8, in update_content_types
> netbox               |     ContentType.objects.filter(app_label='core', model='objectchange').delete()
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/models/query.py", line 1188, in delete
> netbox               |     deleted, _rows_count = collector.delete()
> netbox               |                            ^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/db/models/deletion.py", line 463, in delete
> netbox               |     signals.pre_delete.send(
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django/dispatch/dispatcher.py", line 189, in send
> netbox               |     response = receiver(signal=self, sender=sender, **named)
> netbox               |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/netbox/core/signals.py", line 121, in handle_deleted_object
> netbox               |     validators = get_config().PROTECTION_RULES.get(model_name, [])
> netbox               |                  ^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/netbox/netbox/config/__init__.py", line 28, in get_config
> netbox               |     _thread_locals.config = Config()
> netbox               |                             ^^^^^^^^
> netbox               |   File "/opt/netbox/netbox/netbox/config/__init__.py", line 48, in __init__
> netbox               |     self._populate_from_cache()
> netbox               |   File "/opt/netbox/netbox/netbox/config/__init__.py", line 71, in _populate_from_cache
> netbox               |     self.config = cache.get('config') or {}
> netbox               |                   ^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 92, in get
> netbox               |     value = self._get(key, default, version, client)
> netbox               |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/cache.py", line 36, in _decorator
> netbox               |     raise e.__cause__
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/django_redis/client/default.py", line 258, in get
> netbox               |     value = client.get(key)
> netbox               |             ^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/commands/core.py", line 1821, in get
> netbox               |     return self.execute_command("GET", name)
> netbox               |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/client.py", line 545, in execute_command
> netbox               |     conn = self.connection or pool.get_connection(command_name, **options)
> netbox               |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/connection.py", line 1074, in get_connection
> netbox               |     connection.connect()
> netbox               |   File "/opt/netbox/venv/lib/python3.12/site-packages/redis/connection.py", line 283, in connect
> netbox               |     raise ConnectionError(self._error_message(e))
> netbox               | redis.exceptions.ConnectionError: Error -2 connecting to redis-cache:6379. Name or service not known.
> netbox               |   Applying extras.0117_move_objectchange...
> ```
> 
> and this errors from redis and redis-cache containers:
> 
> ```
> netbox-redis         | 1:M 08 Sep 2024 16:11:22.284 * Reading RDB base file on AOF loading...
> netbox-redis         | 1:M 08 Sep 2024 16:11:22.284 # Can't handle RDB format version 12
> netbox-redis         | 1:M 08 Sep 2024 16:11:22.284 # Error reading the RDB base file appendonly.aof.65.base.rdb, AOF loading aborted
> ```
> 
> After purge volumes from this containers and and re-removal 0011_move_objectchange from django_migrations netbox is up and running:
> 
> ```
> netbox               | Running migrations:
> netbox               |   Applying core.0011_move_objectchange... OK
> netbox               |   Applying extras.0117_move_objectchange... OK
> netbox               |   Applying extras.0118_customfield_uniqueness... OK
> netbox               |   Applying extras.0119_notifications... OK
> netbox               |   Applying circuits.0044_circuit_groups... OK
> netbox               |   Applying core.0012_job_object_type_optional... OK
> netbox               |   Applying extras.0120_eventrule_event_types... OK
> netbox               |   Applying extras.0121_customfield_related_object_filter... OK
> netbox               |   Applying dcim.0188_racktype... OK
> netbox               |   Applying dcim.0189_moduletype_rack_airflow... OK
> netbox               |   Applying dcim.0190_nested_modules... OK
> netbox               |   Applying ipam.0070_vlangroup_vlan_id_ranges... OK
> netbox               |   Applying virtualization.0039_virtualmachine_serial_number... OK
> netbox               |   Applying virtualization.0040_convert_disk_size... OK
> netbox               |   Applying wireless.0009_wirelesslink_distance... OK
> ```
> 
> Thank you very much!

Had to run through the same steps on our Netbox 4.0.8 to 4.1.4 upgrade. Thanks for the tips, it helped us out!

**Comment by FrankPGH** [comment]
I was able to upgrade to 4.1.1 from 4.0.10 in the following way:

(FYI I always snapshot my VM prior to upgrades, so I can just roll back if it fails. The presumption is that each attempt is from a current working state, not a broken upgrade)

If I attempted the upgrade as documented, I got the error:

_psycopg.errors.InsufficientPrivilege: permission denied for schema public_

I was not able to proceed in any manner from this point. 

After a rollback, before _sudo git pull origin master_ :

```
$ sudo -u postgres psql
postgres=# ALTER DATABASE netbox OWNER TO netbox;
ALTER DATABASE
postgres=# \connect netbox;
postgres=# GRANT ALL ON SCHEMA public TO netbox;
GRANT
```

Then proceed as normal and the upgrade completed successfully. I don't know why permissions on the database was suddenly a problem, but if the permissions weren't fixed prior to running upgrade.sh, the whole thing was broken beyond my ability to repair.

**Reply by nerdheim** [reply]
Thanks! I were stuck up updage (native in Arch VM) with the same problem for over a month and could not update it.
This fixed it for me too ^^
